.github/workflows/ci.yaml

```diff
@@ -117,7 +117,15 @@ jobs:
             ${{ runner.os }}-build-
             ${{ runner.os }}-
       - run: pip install black[d]==22.12.0
-      - run: cargo test --all
+      - run: |
+          export INSTA_SNAPSHOT_REFERENCES_FILE="$(mktemp)"
+          cargo test --all
+          UNUSED_SNAPSHOTS=$(cat <(git ls-files '*.snap' | xargs realpath) <(cat $INSTA_SNAPSHOT_REFERENCES_FILE) | sort | uniq -u)
+          if [ ! -z "$UNUSED_SNAPSHOTS" ]; then
+            echo "Found unused snapshots:"
+            echo "$UNUSED_SNAPSHOTS"
+            exit 1
+          fi
```

Comment:
When unused snapshots exist:

https://github.com/charliermarsh/ruff/actions/runs/3811933348/jobs/6484869555

Reply:
The other option would be to run `cargo insta test --delete-unreferenced-snapshots` then `git diff`... but that requires running tests twice.

Reply:
Forgive the drive-by but I came across this — another approach is running `cargo insta test --delete-unreferenced-snapshots` as the main test command, and then doing `git diff` — then tests only need to run once. And `cargo insta test` covers all tests.

Reply:
@max-sixty Thanks for the comment. I thought `cargo insta test` only runs snapshot tests but looks like I was wrong. I'll file a RP :)